packages/bff/src/model/domain/errors.ts

```diff
@@ -182,3 +187,54 @@ export function eserviceRiskNotFound(
     title: "Risk analysis not found",
   });
 }
+
+export function missingActivePurposeVersion(
+  purposeId: string
+): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `There is no active version for purpose ${purposeId}`,
+    code: "missingActivePurposeVersion",
+    title: "Missing active purpose version",
+  });
+}
+
+export function activeAgreementByEserviceAndConsumerNotFound(
+  eserviceId: string,
+  consumerId: string
+): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `Active agreement for Eservice ${eserviceId} and consumer ${consumerId} not found`,
+    code: "activeAgreementByEserviceAndConsumerNotFound",
+    title: "Active agreement not found",
+  });
+}
+
+export function multipleAgreementForEserviceAndConsumer(
+  eserviceId: string,
+  consumerId: string
+): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `Multiple agreements for Eservice ${eserviceId} and consumer ${consumerId} found`,
+    code: "multipleAgreementForEserviceAndConsumer",
+    title: "Multiple agreement found",
+  });
+}
+
+export function purposeIdNotFoundInClientAssertion(): ApiError<ErrorCodes> {
+  return new ApiError({
+    detail: `PurposeId not found in client assertion`,
```

Comment:
Maybe log the `purposeId` to make the log more informative?

Reply:
Well i cannot because, the id is undefined :)

Reply:
True that 😅  My bad, I read purpose not found and missed the Id